packages/auth/src/AuthProvider.tsx

```diff
@@ -1,24 +1,276 @@
+        if (!codeDeliveryDetails) {
+          setAuthenticated(true);
+        }
+
+        user =
+          (await Auth.currentAuthenticatedUser()) as CognitoUserWithAttributes;
+      } catch (err) {
+        const e = err as { code?: string; message?: string };
+        if (!e?.code) {
+          return {
+            action,
+            error: e.message || "Unknown Error",
+          };
+        }
+
+        error = e.code;
+
+        if (e.code === SIGN_IN_ERROR_CODES.UserNotFoundException) {
+          const result = await Auth.signUp({
+            username: phone,
+            password: hashedPassword,
+            attributes: {
+              phone_number: phone,
+            },
+          });
+
+          codeDeliveryDetails = result.codeDeliveryDetails;
+          user = result.user;
+        } else if (e.code === SIGN_IN_ERROR_CODES.UserNotConfirmedException) {
+          const result = await Auth.resendSignUp(phone);
+
+          codeDeliveryDetails = result.codeDeliveryDetails;
+        }
+      }
+
+      setCognitoUser(user);
+
+      return {
+        user,
+        codeDeliveryDetails: codeDeliveryDetails,
+        error,
+        action,
+      };
+    },
+    [
+      Auth,
+      setAuthenticated,
+      setCognitoUser,
+      setCognitoUserSignIn,
+      SIGN_IN_OR_CREATE_ACTIONS,
+    ]
+  );
+
+  const getPassword = (phoneNumber: string) => MD5(`${phoneNumber}`).toString();
+
+  const updateUserAttributes = useCallback(
+    async (data) => {
+      await Auth.updateUserAttributes(cognitoUser, data);
+      const newCognitoUser = await Auth.currentAuthenticatedUser({
+        bypassCache: false, // Optional, By default is false. If set to true, this call will send a request to Cognito to get the latest user data
+      });
+      setCognitoUser(newCognitoUser);
+    },
+    [Auth, cognitoUser]
+  );
+
+  const confirmSignUp = useCallback(
+    async (phoneNumber, answer) => {
+      try {
+        await Auth.confirmSignUp(phoneNumber, answer);
+        await Auth.signIn(phoneNumber, getPassword(phoneNumber));
+        const user = await Auth.currentAuthenticatedUser();
+        await Auth.setPreferredMFA(user, "SMS");
+        setCognitoUser(user);
+        setAuthenticated(true);
+        return { success: true };
+      } catch (e) {
+        if (e === "No current user") {
+          return {
+            success: false,
+            error: ANSWER_CHALLENGE_ERRORS.INCORRECT_CODE,
+          };
+        }
+        throw e;
+      }
+    },
+    [cognitoUser, Auth]
+  );
+
+  const resendSignUp = useCallback(
+    async (phoneNumber: string) => {
+      await Auth.resendSignUp(phoneNumber);
+    },
+    [Auth]
   );
+
+  const confirmSignIn = useCallback(
+    async (answer: string) => {
+      try {
+        await Auth.confirmSignIn(cognitoUserSignIn, answer);
+        const user = await Auth.currentAuthenticatedUser();
+        setCognitoUser(user);
```

Comment:
sign in should also check if MFA is set, if MFA is disabled it should enable it for the user

Reply:
Updated

Our config currently don't allow signing in when phone number is not verified so they will always be caught by confirmSignUp (and have their MFA set to SMS) but just in case somehow their preferredMFA is turned off, we now able to check it on login and update it to use SM. We also asked them to sign in again to validate them using SMS OTP.

https://github.com/applyinnovations/aws-amplify-hooks/pull/99/files#diff-aa11efa890050e850cb833101c824d44d8528da61076e5119c6928effebd063fR130